compiler/rustc_codegen_ssa/src/mir/block.rs

```diff
@@ -220,15 +220,8 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
             bx.cleanup_ret(funclet, None);
         } else {
             let slot = self.get_personality_slot(&mut bx);
-            let lp0 = slot.project_field(&mut bx, 0);
-            let lp0 = bx.load_operand(lp0).immediate();
-            let lp1 = slot.project_field(&mut bx, 1);
-            let lp1 = bx.load_operand(lp1).immediate();
+            let lp = bx.load(self.landing_pad_type(), slot.llval, slot.align);
             slot.storage_dead(&mut bx);
-
-            let mut lp = bx.const_undef(self.landing_pad_type());
-            lp = bx.insert_value(lp, lp0, 0);
-            lp = bx.insert_value(lp, lp1, 1);
             bx.resume(lp);
```

Comment:
I would actually prefer if `resume` would take an array. That would reduce one place where cg_ssa assumes that a `Value` can hold structs, rather than just primitives. In Cranelift a `Value` can only contain integers, floats, pointers and vectors. It doesn't support complex types.